src/main.js

```diff
@@ -0,0 +1,11 @@
+import {todoInput} from "./components/todoInput.js";
+import {todoCheckbox} from "./components/todoCheckbox.js";
+import {todoEdit} from "./components/todoEdit.js";
+import {todoCount} from "./components/todoCount.js";
+import {todoFliter} from "./components/todoFilter.js";
+
+todoInput();
+todoCheckbox();
+todoEdit();
+todoCount();
+todoFliter();
```

Comment:
모듈로 불러온 함수들도 하나의 모듈로 만들어서 관리하면 더 좋을거 같습니다!

```suggestion
const App = () => {
  todoInput();
  todoCheckbox();
  todoEdit();
  todoCount();
  todoFliter();
}

new App();
```

Reply:
윤호님 코드리뷰 감사합니다!    
모듈화에는 윤호님이랑 같이 했던 페어프로그래밍이 도움이 되었습니다 👍  

제안해주신 코드에서 마지막에 new App();은 왜 쓰인 건지 알려주실 수 있을까요? 여기서 new가 어떤 역할인지 모르겠어서요

Reply:
지선님이 사용하신 코드에서는 바로 실행을 시켜서 ```App```이라는 모듈이 필요없을거 같습니다! 

다만 만약에 제가 수정한 코드처럼 ```App```이라는 모듈 안에 지선님이 호출하신 함수들이 들어간다면 최초에 window가 load 됐을 때 모듈화 된 함수를 실행시켜줄 필요가 있어서 위와 같이 ```new```라는 연산자를 사용해서 메인 모듈을 실행시켜야 이벤트 리스너 등록 및 지선님이 원하시는 작동이 정상적으로 일어날 수 있습니다!

아래의 링크를 참고해보시면 좋을거 같습니다! 다만 앞으로의 범용성을 생각한다면 Main이 되는 함수도 모듈화 시키는게 좋을거 같다는건 제 개인적인 의견이라 리뷰 남겨주시는 다른 분들의 의견도 참고하시면 좋을거 같습니다!

고생하셨습니다!!👍👍👍

- [new operator / MDN (링크)](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/new)

Reply:
설명 감사합니다! 의견 주신 부분 반영해보도록 하겠습니다 👏

Reply:
오 저도 궁금하고,, 정확히는 반대 의견을 갖고있습니다. 
제가 이해하기로는 jiseonk님의 코드는 마크업에 기능을 입히는 함수들, 즉 사용자의 동작에 따라 화면상에 다른 요소를 표현하는 코드( 그리고 그 함수가 사용하는 함수들 )이었습니다. 제가 어느정도 코드들의 역할과 문맥을 이해했다고 한다면, 아래와 같은 이유로 new 사용은 적절치 않다는 의견 드립니다..

따로 화면에 표시되는 데이터 (todo들, 필터링상태)는 관리하고 있지 않은 코드이고, 그렇기 때문에, `main.js`는 마크업에 기능을 입히는 함수들만 실행해주는 역할을 하고 있습니다.

단순히 함수들을 실행하는 함수를 new로 호출한다면, (내부적으로 [Call] 내부메서드가 아닌 [Construct] 내부메서드가 호출되어) prototype을 생성하고, 인스턴스도 생성하게 되는데 이러한 동작이 문맥상 불필요해 보입니다. window 라는 브라우저가 제공하는 호스트 객체에 프로퍼티를 추가할 이유도 없어보입니다.

Reply:
아하 그런 관점에서 생각해보지 못했는데 동욱님 말씀이 더 맞는거 같다는 생각이 드네요! 

```window```객체에 굳이 프로퍼티를 추가할 필요는 없어보입니다! 

제가 관성적으로 저런식으로 많이 사용을 해서 주관에 빗댄 의견을 드렸네요! ```new```로 호출했을 때 어떻게 동작하는지도 다시 한번 찾아봐야겠습니다! 감사합니다 동욱님!!👍👍👍

Reply:
두분 다 의견 감사합니다! 덕분에 코드를 어떻게 짜는 게 좋을지 여러가지로 생각해볼 수 있었습니다 👍 저도 new 관련 문법을 좀 더 찾아봐야겠네요.